src/gpu/intel/compute/kernel_ctx.hpp

```diff
@@ -52,13 +52,7 @@ class kernel_ctx_t {
         for (auto &opt : option_set_)
             oss << " " << opt;
 
-        if (use_int32_offset_) {
-            oss << " -DUSE_INT32_OFFSET";
-        } else {
-            // TODO: Determine if specialization for buffers between 2GB and 4GB
-            // is worthwhile
-            oss << " -cl-intel-greater-than-4GB-buffer-required";
-        }
+        if (use_int32_offset_) { oss << " -DUSE_INT32_OFFSET"; }
```

Comment:
It looks like the logic to generate `-cl-intel-greater-than-4GB-buffer-required` was dropped.